diff --git a/sections/cookie-banner.liquid b/sections/cookie-banner.liquid
index b08f572..a76ce23 100644
--- a/sections/cookie-banner.liquid
+++ b/sections/cookie-banner.liquid
@@ -8,13 +8,18 @@
     --color-highlight: {{ section.settings.color_drawer_highlight.rgb }};
   }

-  #shopify-pc__banner {
+  #shopify-pc__banner,
+  #shopify-pc__banner[style*="display: block"] {
     display: none !important;
   }
 </style>

-<link rel="stylesheet" href="{{ 'cookie-banner.css' | asset_url }}" media="print" onload="this.media='all'">
-<noscript>{{ 'cookie-banner.css' | asset_url | stylesheet_tag }}</noscript>
+{%- unless request.design_mode -%}
+  <link rel="stylesheet" href="{{ 'cookie-banner.css' | asset_url }}" media="print" onload="this.media='all'">
+  <noscript>{{ 'cookie-banner.css' | asset_url | stylesheet_tag }}</noscript>
+{%- else -%}
+  {{ 'cookie-banner.css' | asset_url | stylesheet_tag }}
+{%- endunless -%}
 <script src="{{ 'cookie-banner.js' | asset_url }}" defer="defer"></script>

 <cookie-banner id="CookieBanner" class="x-modal cookie-modal drawer z-35 fixed bottom-0 left-0 h-full w-full pointer-events-none"
@@ -28,7 +33,7 @@
 >
   <overlay-element class="overlay fixed-modal invisible opacity-0 fixed bottom-0 left-0 w-full h-screen pointer-events-none" aria-controls="CookieBanner" aria-expanded="false"></overlay-element>
   <div class="drawer__inner z-10 absolute bottom-0 {{ section.settings.content_position }} flex flex-col w-full h-full overflow-hidden">
-    <gesture-element class="drawer__header flex justify-between opacity-0 invisible relative sr-only">
+    <gesture-element class="drawer__header flex justify-between opacity-0 invisible relative sr-only" tabindex="0">
       <button class="button button--secondary button--close drawer__close hidden sm:flex items-center justify-center" type="button" is="hover-button" aria-controls="AgeVerifier" aria-expanded="false" aria-label="{{ 'general.accessibility.close' | t | escape }}">
         <span class="btn-fill" data-fill></span>
         <span class="btn-text">
@@ -50,14 +55,14 @@
           {%- for block in section.blocks -%}
             {%- case block.type -%}
               {%- when 'spacing' -%}
-                <empty-element class="block spacing-section{% if block.settings.height_mobile == 0 %} hidden lg:block{% endif %}{% if block.settings.height == 0 %} lg:hidden{% endif %}" style="--spacing-height-desktop: {{ block.settings.height }}px; --spacing-height-mobile: {{ block.settings.height_mobile }}px;" {{ block.shopify_attributes }}></empty-element>
+                <empty-element class="block spacing-section{% if block.settings.height_mobile == 0 %} hidden lg:block{% endif %}{% if block.settings.height == 0 %} lg:hidden{% endif %}" style="--spacing-height-desktop: {{ block.settings.height }}px;--spacing-height-mobile: {{ block.settings.height_mobile }}px;" {{ block.shopify_attributes }}></empty-element>

               {%- when 'heading' -%}
                 {%- if block.settings.heading != blank -%}
                   {%- assign delay = delay | plus: 1 -%}
                   <{{ block.settings.heading_tag }} id="Dialog-{{ section.id }}-Heading{{ delay }}" class="heading leading-none tracking-none {{ block.settings.heading_size }}" {{ block.shopify_attributes }}>
                     {%- if block.settings.highlighted_text == 'none' -%}
-                      <split-words class="split-words block" data-animate="fade-up-large" data-animate-delay="{{ delay | times: 250 }}">{{ block.settings.heading }}</split-words>
+                      <split-words class="split-words js-invisible block" data-animate="fade-up-large" data-animate-delay="{{ delay | times: 250 }}">{{ block.settings.heading }}</split-words>
                     {%- else -%}
                       {%- render 'highlighted-text', content: block.settings.heading, style: block.settings.highlighted_text, scribble: block.settings.highlighted_scribble, gradient: with_gradient -%}
                     {%- endif -%}
@@ -66,14 +71,14 @@

               {%- when 'text' -%}
                 {%- if block.settings.text != blank -%}
-                  <div class="rte leading-normal {{ block.settings.text_font }} {{ block.settings.text_size }}{% if block.settings.secondary_color %} text-opacity{% endif %}"{% if block.settings.text_size == 'subtext-custom' %} style="--subtext-size:{{ block.settings.text_max_size }}px;"{% endif %} {{ block.shopify_attributes }}>
+                  <div class="rte {{ block.settings.text_font }} {{ block.settings.text_size }}{% if block.settings.text_font == 'subtitle' %} uppercase leading-tight tracking-widest{% else %} leading-normal{% endif %}{% if block.settings.secondary_color %} text-opacity{% endif %}"{% if block.settings.text_size == 'subtext-custom' %} style="--subtext-size: {{ block.settings.text_max_size }}px;--subtext-line-height: {{ block.settings.text_line_height }};--subtext-letter-spacing: {{ block.settings.text_letter_spacing | divided_by: 1000.00 }}em;{% if block.settings.text_capitalize %}--subtext-text-transform: uppercase;{% endif %}"{% endif %} {{ block.shopify_attributes }}>
                     {%- render 'highlighted-text', content: block.settings.text, style: block.settings.highlighted_text, scribble: block.settings.highlighted_scribble, gradient: with_gradient -%}
                   </div>
                 {%- endif -%}

               {%- when 'buttons' -%}
                 <div class="cookie__buttons" {{ block.shopify_attributes }}>
-                  <div class="inline-flex gap-4">
+                  <div class="cookie__flex flex flex-wrap gap-4">
                     {%- if block.settings.button_label_1 != blank -%}
                       <button type="button" name="accept" class="button button--primary icon-with-text" is="hover-button">
                         <span class="btn-fill" data-fill></span>
@@ -100,18 +105,42 @@
                 </div>

               {%- when 'icon' -%}
-                {%- if block.settings.icon_image != blank -%}
-                  <div class="leading-none" {{ block.shopify_attributes }}>
-                    <figure class="media media--transparent relative inline-block">
-                      {%- capture sizes -%}{{ block.settings.icon_width }}px{%- endcapture -%}
-                      {%- capture widths -%}{{ block.settings.icon_width }}, {{ block.settings.icon_width | times: 2 | at_most: block.settings.icon_image.width }}{%- endcapture -%}
-                      {{- block.settings.icon_image | image_url: width: block.settings.icon_image.width | image_tag: loading: 'lazy', sizes: sizes, widths: widths, class: 'image-fit' -}}
-                    </figure>
+                {%- if block.settings.icon != 'none' or block.settings.icon_image != blank -%}
+                  {%- capture icon_content -%}
+                    {%- if block.settings.icon_image != blank -%}
+                      {%- liquid
+                        if block.settings.iwidth == 'fit'
+                          assign icon_width = block.settings.icon_image.width
+                        else
+                          assign icon_width = block.settings.icon_width
+                        endif
+                      -%}
+                      <figure class="media media--transparent relative inline-block" {{ block.shopify_attributes }}>
+                        {%- capture sizes -%}{{ icon_width }}px{%- endcapture -%}
+                        {%- capture widths -%}{{ icon_width }}, {{ icon_width | times: 2 | at_most: block.settings.icon_image.width }}{%- endcapture -%}
+                        {%- assign image_alt = block.settings.icon_image.alt | default: block.settings.accessibility_info | escape -%}
+                        {{- block.settings.icon_image | image_url: width: block.settings.icon_image.width | image_tag: loading: 'lazy', sizes: sizes, widths: widths, alt: image_alt, class: 'image-fit' -}}
+                      </figure>
+                    {%- elsif block.settings.icon != 'none' -%}
+                      {%- liquid
+                        if block.settings.iwidth == 'fit'
+                          render 'icon-guarantee', icon: block.settings.icon, size: 'lg', class: 'inline-block', shopify_attributes: block.shopify_attributes
+                        else
+                          render 'icon-guarantee', icon: block.settings.icon, size: 'custom', class: 'inline-block', width: block.settings.icon_width, shopify_attributes: block.shopify_attributes
+                        endif
+                      -%}
+                    {%- endif -%}
+                  {%- endcapture -%}
+
+                  <div class="leading-none">
+                    {%- if block.settings.link != blank -%}
+                      <a class="inline-block" href="{{ block.settings.link }}"{% if block.settings.external %} target="_blank" rel="noopener"{% endif %} aria-label="{{ block.settings.accessibility_info | escape }}">
+                        {{- icon_content -}}
+                      </a>
+                    {%- else -%}
+                      {{- icon_content -}}
+                    {%- endif -%}
                   </div>
-                {%- elsif block.settings.icon != 'none' -%}
-                  <p class="leading-none" {{ block.shopify_attributes }}>
-                    {%- render 'icon-guarantee', icon: block.settings.icon, size: 'lg', class: 'inline-block' -%}
-                  </p>
                 {%- endif -%}

               {%- when 'image' -%}
@@ -362,6 +391,7 @@
           "label": "t:sections.global.settings.highlighted_text.label"
         },
         {
+          "visible_if": "{{ block.settings.highlighted_text == 'scribble' }}",
           "type": "select",
           "id": "highlighted_scribble",
           "options": [
@@ -402,6 +432,26 @@
           "default": "<p>We use cookies and similar technologies to provide the best experience on our website. <a href=\"/policies/privacy-policy\">Privacy Policy</a></p>",
           "label": "t:sections.global.settings.text.label"
         },
+        {
+          "type": "select",
+          "id": "text_font",
+          "options": [
+            {
+              "value": "body",
+              "label": "t:sections.global.settings.text_font.options__1.label"
+            },
+            {
+              "value": "heading",
+              "label": "t:sections.global.settings.text_font.options__2.label"
+            },
+            {
+              "value": "subtitle",
+              "label": "t:sections.global.settings.text_font.options__3.label"
+            }
+          ],
+          "default": "body",
+          "label": "t:sections.global.settings.text_font.label"
+        },
         {
           "type": "select",
           "id": "text_size",
@@ -431,6 +481,7 @@
           "label": "t:sections.global.settings.text_size.label"
         },
         {
+          "visible_if": "{{ block.settings.text_size == 'subtext-custom' }}",
           "type": "range",
           "id": "text_max_size",
           "default": 32,
@@ -441,20 +492,30 @@
           "label": "t:sections.global.settings.text_max_size.label"
         },
         {
-          "type": "select",
-          "id": "text_font",
-          "options": [
-            {
-              "value": "body",
-              "label": "t:sections.global.settings.text_font.options__1.label"
-            },
-            {
-              "value": "heading",
-              "label": "t:sections.global.settings.text_font.options__2.label"
-            }
-          ],
-          "default": "body",
-          "label": "t:sections.global.settings.text_font.label"
+          "visible_if": "{{ block.settings.text_size == 'subtext-custom' }}",
+          "type": "range",
+          "id": "text_line_height",
+          "label": "t:sections.global.settings.text_line_height.label",
+          "default": 1.6,
+          "min": 0.8,
+          "max": 2,
+          "step": 0.1
+        },
+        {
+          "visible_if": "{{ block.settings.text_size == 'subtext-custom' }}",
+          "type": "range",
+          "id": "text_letter_spacing",
+          "label": "t:sections.global.settings.text_letter_spacing.label",
+          "default": 0,
+          "min": -75,
+          "max": 250,
+          "step": 5
+        },
+        {
+          "visible_if": "{{ block.settings.text_size == 'subtext-custom' }}",
+          "type": "checkbox",
+          "id": "text_capitalize",
+          "label": "t:sections.global.settings.text_capitalize.label"
         },
         {
           "type": "checkbox",
@@ -504,6 +565,7 @@
           "label": "t:sections.global.settings.highlighted_text.label"
         },
         {
+          "visible_if": "{{ block.settings.highlighted_text == 'scribble' }}",
           "type": "select",
           "id": "highlighted_scribble",
           "options": [
@@ -563,7 +625,7 @@
           "type": "select",
           "id": "icon",
           "label": "t:sections.global.settings.icon.label",
-          "default": "none",
+          "default": "tag",
           "options": [
             {
               "value": "none",
@@ -800,15 +862,50 @@
           "id": "icon_image",
           "label": "t:sections.global.settings.icon_image.label"
         },
+        {
+          "type": "select",
+          "id": "iwidth",
+          "label": "t:sections.global.settings.icon_width.label",
+          "options": [
+            {
+              "value": "fit",
+              "label": "t:sections.global.settings.icon_width.options__1.label"
+            },
+            {
+              "value": "custom",
+              "label": "t:sections.global.settings.icon_width.options__2.label"
+            }
+          ],
+          "default": "custom"
+        },
         {
           "type": "range",
           "id": "icon_width",
-          "label": "t:sections.global.settings.icon_width.label",
+          "label": "t:sections.global.settings.custom_width.label",
           "default": 48,
           "min": 10,
-          "max": 120,
+          "max": 200,
           "step": 2,
-          "unit": "px"
+          "unit": "px",
+          "visible_if": "{{ block.settings.iwidth == 'custom' }}"
+        },
+        {
+          "type": "url",
+          "id": "link",
+          "label": "t:sections.global.settings.link.label"
+        },
+        {
+          "type": "text",
+          "id": "accessibility_info",
+          "label": "t:sections.global.settings.accessibility.label",
+          "info": "t:sections.global.settings.accessibility.info",
+          "visible_if": "{{ block.settings.link != blank }}"
+        },
+        {
+          "type": "checkbox",
+          "id": "external",
+          "label": "t:sections.global.settings.button_external.label",
+          "default": false
         }
       ]
     },
@@ -838,6 +935,7 @@
           "label": "t:sections.global.settings.image_width.label"
         },
         {
+          "visible_if": "{{ block.settings.image_width == 'custom' }}",
           "type": "range",
           "id": "image_max_width",
           "label": "t:sections.global.settings.image_max_width.label",
@@ -848,6 +946,7 @@
           "unit": "px"
         },
         {
+          "visible_if": "{{ block.settings.image_width == 'custom' }}",
           "type": "range",
           "id": "image_max_width_mobile",
           "label": "t:sections.global.settings.image_max_width_mobile.label",
