diff --git a/sections/header.liquid b/sections/header.liquid
index 4b61a43..702a8ed 100644
--- a/sections/header.liquid
+++ b/sections/header.liquid
@@ -5,67 +5,7 @@
     --color-background: {{ section.settings.color_background.rgb }};
     --color-foreground: {{ section.settings.color_text.rgb }};
     --color-transparent: {{ section.settings.transparent_color_text.rgb }};
-  }
-
-  /* Header menu responsive styles */
-  .header__menu > ul {
-    display: flex;
-    flex-wrap: nowrap;
-    min-width: 0;
-    width: 100%;
-  }
-
-  .header__menu > ul > li {
-    flex-shrink: 1;
-  }
-
-  .header__menu > ul .menu__item,
-  .single-custom-url {
-    white-space: nowrap;
-  }
-
-  /* Large screens */
-  @media screen and (min-width: 1281px) {
-    .header__menu > ul.with-block .menu__item,
-    .single-custom-url {
-      padding-left: 1rem;
-      padding-right: 1rem;
-      font-size: 1rem;
-    }
-  }
-
-  /* Medium-large screens */
-  @media screen and (max-width: 1280px) {
-    .header__menu > ul.with-block .menu__item,
-    .single-custom-url {
-      padding-left: 0.75rem;
-      padding-right: 0.75rem;
-      font-size: 0.875rem;
-    }
-  }
-
-  /* Medium screens */
-  @media screen and (max-width: 1140px) {
-    .header__menu > ul.with-block .menu__item,
-    .single-custom-url {
-      padding-left: 0.5rem;
-      padding-right: 0.5rem;
-      font-size: 0.8125rem;
-    }
-  }
-
-  /* Smaller screens */
-  @media screen and (max-width: 1074px) {
-    .header__menu > ul.with-block .menu__item,
-    .single-custom-url {
-      padding-left: 0.375rem;
-      padding-right: 0.375rem;
-      font-size: 0.85rem;
-    }
-
-    .header__menu > ul > li {
-      flex: 0 1 auto;
-    }
+    --color-localization: {{ section.settings.color_localization.rgb }};
   }
 </style>

@@ -97,40 +37,40 @@

 <header data-section-id="{{ section.id }}" class="header header--{{ header_layout }} mobile:header--{{ section.settings.logo_position_mobile }} page-width{% if section.settings.full_width %} page-width--full{% endif %} section{% if section.settings.rounded %} section--rounded{% endif %} section--padding grid items-center z-20 relative w-full"{% if section.settings.sticky_header_type != 'none' %} is="sticky-header" data-sticky-type="{{ section.settings.sticky_header_type }}"{% else %} is="custom-header"{% endif %}>
   {%- liquid
-    render 'header-icons', show_search: true, show_hamburger: true, show_toggle: show_toggle, dir: 'start'
+    render 'header-icons', section: section, show_search: true, show_hamburger: true, show_toggle: show_toggle, context: 'start'

     if header_layout != 'center-left' and header_layout != 'center-split'
-      render 'header-logo'
+      render 'header-logo', logo_h1: section.settings.logo_h1
     endif

     if header_layout == 'center'
-      render 'header-icons', show_search: true, show_account: true, show_cart: true, show_hamburger: true, dir: 'end'
+      render 'header-icons', section: section, show_search: true, show_account: true, show_cart: true, show_hamburger: true, context: 'end'
     endif
   -%}

   <div class="header__navigation hidden lg:flex lg:gap-5 {{ main_menu_class }}"{% if header_layout == 'center' %} id="MenuToggle"{% endif %}>
     <nav class="header__menu hidden lg:flex" role="navigation" aria-label="Primary">
-      {%- render 'header-nav-desktop', section: section, main_menu: main_menu, limit: link_count -%}
+      {%- render 'header-nav-desktop', section: section, main_menu: main_menu, limit: link_count, context: 'start' -%}
     </nav>
   </div>

   {%- liquid
     if header_layout == 'center-left' or header_layout == 'center-split'
-      render 'header-logo'
+      render 'header-logo', logo_h1: section.settings.logo_h1
     endif
   -%}

   {%- if header_layout == 'center-split' -%}
     <div class="header__navigation header__navigation--split hidden lg:flex lg:gap-5 lg:justify-start">
       <nav class="header__menu hidden lg:flex" role="navigation" aria-label="Secondary">
-        {%- render 'header-nav-desktop', section: section, main_menu: main_menu, offset: link_count -%}
+        {%- render 'header-nav-desktop', section: section, main_menu: main_menu, offset: link_count, context: 'end' -%}
       </nav>
     </div>
   {%- endif -%}

   {%- liquid
     if header_layout != 'center'
-      render 'header-icons', show_search: true, show_account: true, show_cart: true, show_hamburger: true, dir: 'end'
+      render 'header-icons', section: section, show_search: true, show_account: true, show_cart: true, show_hamburger: true, context: 'end'
     endif
   -%}

@@ -147,11 +87,11 @@
   <overlay-element class="overlay opacity-0 z-1 fixed left-0 w-full pointer-events-none"></overlay-element>
 {%- endif -%}

-<script>
-  if (Shopify.designMode) {
-    document.documentElement.style.setProperty('--header-height', Math.round(document.getElementById('shopify-section-{{ section.id }}').clientHeight) + 'px');
-  }
-</script>
+{%- if request.design_mode -%}
+  <script>
+    document.documentElement.style.setProperty('--header-height', document.getElementById('shopify-section-{{ section.id }}').getBoundingClientRect().height.toFixed(1) + 'px');
+  </script>
+{%- endif -%}

 {% schema %}
 {
@@ -162,6 +102,13 @@
       "type": "paragraph",
       "content": "t:sections.header.settings.paragraph__logo_help.content"
     },
+    {
+      "type": "checkbox",
+      "id": "logo_h1",
+      "default": true,
+      "label": "t:sections.header.settings.logo_h1.label",
+      "info": "t:sections.header.settings.logo_h1.info"
+    },
     {
       "type": "link_list",
       "id": "menu",
@@ -175,6 +122,7 @@
       "info": "t:sections.header.settings.menu_mobile.info"
     },
     {
+      "visible_if": "{{ section.settings.header_layout != 'left-drawer' and section.settings.header_layout != 'center-drawer' }}",
       "type": "select",
       "id": "menu_active",
       "label": "t:sections.header.settings.menu_active.label",
@@ -191,6 +139,7 @@
       ]
     },
     {
+      "visible_if": "{{ section.settings.header_layout != 'left-drawer' and section.settings.header_layout != 'center-drawer' }}",
       "type": "checkbox",
       "id": "hover_menu",
       "default": true,
@@ -252,6 +201,36 @@
       "default": "always",
       "label": "t:sections.header.settings.sticky_header_type.label"
     },
+    {
+      "type": "header",
+      "content": "t:sections.header.settings.header__icons.content"
+    },
+    {
+      "type": "select",
+      "id": "icons_size",
+      "options": [
+        {
+          "value": "sm",
+          "label": "t:sections.header.settings.icons_size.options__1.label"
+        },
+        {
+          "value": "md",
+          "label": "t:sections.header.settings.icons_size.options__2.label"
+        },
+        {
+          "value": "lg",
+          "label": "t:sections.header.settings.icons_size.options__3.label"
+        }
+      ],
+      "default": "lg",
+      "label": "t:sections.header.settings.icons_size.label"
+    },
+    {
+      "visible_if": "{{ section.settings.header_layout == 'left-drawer' or section.settings.header_layout == 'center-drawer' }}",
+      "type": "text",
+      "id": "menu_drawer_icon_label",
+      "label": "t:sections.header.settings.menu_drawer_icon_label.label"
+    },
     {
       "type": "header",
       "content": "t:sections.header.settings.header__language_selector.content",
@@ -275,12 +254,14 @@
       "default": false
     },
     {
+      "visible_if": "{{ section.settings.enable_country_selector_desktop == true }}",
       "type": "checkbox",
       "id": "show_country_name_dekstop",
       "label": "t:sections.header.settings.show_country_name.label",
       "default": false
     },
     {
+      "visible_if": "{{ section.settings.enable_country_selector_desktop == true }}",
       "type": "checkbox",
       "id": "show_country_flag_desktop",
       "label": "t:sections.header.settings.show_country_flag.label",
@@ -310,6 +291,12 @@
       "default": "center",
       "label": "t:sections.header.settings.logo_position_mobile.label"
     },
+    {
+      "type": "checkbox",
+      "id": "accordion_menu_mobile",
+      "default": false,
+      "label": "t:sections.header.settings.enable_accordion_menu.label"
+    },
     {
       "type": "checkbox",
       "id": "show_social",
@@ -361,6 +348,12 @@
       "label": "t:sections.global.settings.color_background.label",
       "default": "#ffffff"
     },
+    {
+      "visible_if": "{{ section.settings.enable_language_selector_desktop == true or section.settings.enable_country_selector_desktop == true }}",
+      "type": "color",
+      "id": "color_localization",
+      "label": "t:sections.header.settings.color_localization.label"
+    },
     {
       "type": "header",
       "content": "t:sections.global.settings.header__section.content"
@@ -457,6 +450,7 @@
           "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "select",
           "id": "promo_image_ratio_1",
           "options": [
@@ -485,44 +479,52 @@
           "label": "t:sections.global.settings.image_ratio.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "text",
           "id": "promo_heading_1",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "inline_richtext",
           "id": "promo_subtext_1",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "checkbox",
           "id": "promo_overlay_content_1",
           "default": true,
           "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "url",
           "id": "promo_link_1",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "text",
           "id": "promo_tag_1",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "color",
           "id": "promo_color_text_1",
           "label": "t:sections.global.settings.color_text.label",
           "default": "#ffffff"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "color",
           "id": "promo_color_overlay_1",
           "label": "t:sections.global.settings.color_overlay.label",
           "default": "#000000"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
           "type": "range",
           "id": "promo_overlay_opacity_1",
           "min": 0,
@@ -542,6 +544,7 @@
           "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "select",
           "id": "promo_image_ratio_2",
           "options": [
@@ -570,44 +573,52 @@
           "label": "t:sections.global.settings.image_ratio.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "text",
           "id": "promo_heading_2",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "inline_richtext",
           "id": "promo_subtext_2",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "checkbox",
           "id": "promo_overlay_content_2",
           "default": true,
           "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "url",
           "id": "promo_link_2",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "text",
           "id": "promo_tag_2",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "color",
           "id": "promo_color_text_2",
           "label": "t:sections.global.settings.color_text.label",
           "default": "#ffffff"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "color",
           "id": "promo_color_overlay_2",
           "label": "t:sections.global.settings.color_overlay.label",
           "default": "#000000"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
           "type": "range",
           "id": "promo_overlay_opacity_2",
           "min": 0,
@@ -627,6 +638,7 @@
           "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "select",
           "id": "promo_image_ratio_3",
           "options": [
@@ -655,44 +667,52 @@
           "label": "t:sections.global.settings.image_ratio.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "text",
           "id": "promo_heading_3",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "inline_richtext",
           "id": "promo_subtext_3",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "checkbox",
           "id": "promo_overlay_content_3",
           "default": true,
           "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "url",
           "id": "promo_link_3",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "text",
           "id": "promo_tag_3",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "color",
           "id": "promo_color_text_3",
           "label": "t:sections.global.settings.color_text.label",
           "default": "#ffffff"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "color",
           "id": "promo_color_overlay_3",
           "label": "t:sections.global.settings.color_overlay.label",
           "default": "#000000"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
           "type": "range",
           "id": "promo_overlay_opacity_3",
           "min": 0,
@@ -712,6 +732,7 @@
           "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "select",
           "id": "promo_image_ratio_4",
           "options": [
@@ -740,44 +761,52 @@
           "label": "t:sections.global.settings.image_ratio.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "text",
           "id": "promo_heading_4",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "inline_richtext",
           "id": "promo_subtext_4",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "checkbox",
           "id": "promo_overlay_content_4",
           "default": true,
           "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "url",
           "id": "promo_link_4",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "text",
           "id": "promo_tag_4",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "color",
           "id": "promo_color_text_4",
           "label": "t:sections.global.settings.color_text.label",
           "default": "#ffffff"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "color",
           "id": "promo_color_overlay_4",
           "label": "t:sections.global.settings.color_overlay.label",
           "default": "#000000"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
           "type": "range",
           "id": "promo_overlay_opacity_4",
           "min": 0,
@@ -797,6 +826,7 @@
           "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "select",
           "id": "promo_image_ratio_5",
           "options": [
@@ -825,44 +855,754 @@
           "label": "t:sections.global.settings.image_ratio.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "text",
+          "id": "promo_heading_5",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "inline_richtext",
+          "id": "promo_subtext_5",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "checkbox",
+          "id": "promo_overlay_content_5",
+          "default": true,
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "url",
+          "id": "promo_link_5",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "text",
+          "id": "promo_tag_5",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "color",
+          "id": "promo_color_text_5",
+          "label": "t:sections.global.settings.color_text.label",
+          "default": "#ffffff"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "color",
+          "id": "promo_color_overlay_5",
+          "label": "t:sections.global.settings.color_overlay.label",
+          "default": "#000000"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "range",
+          "id": "promo_overlay_opacity_5",
+          "min": 0,
+          "max": 100,
+          "step": 1,
+          "unit": "%",
+          "label": "t:sections.global.settings.overlay_opacity.label",
+          "default": 40
+        }
+      ]
+    },
+    {
+      "type": "mega_menu_tab",
+      "name": "t:sections.header.blocks.mega_menu_tab.name",
+      "settings": [
+        {
+          "type": "text",
+          "id": "menu_title",
+          "label": "t:sections.header.blocks.mega_menu.settings.menu_title.label",
+          "info": "t:sections.header.blocks.mega_menu.settings.menu_title.info"
+        },
+        {
+          "type": "text",
+          "id": "button_label",
+          "default": "Button label",
+          "label": "t:sections.global.settings.button_label.label",
+          "info": "t:sections.global.settings.button_label.info"
+        },
+        {
+          "type": "url",
+          "id": "button_link",
+          "label": "t:sections.global.settings.button_link.label"
+        },
+        {
+          "type": "text",
+          "id": "heading_1",
+          "default": "Collections",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.heading_1.label"
+        },
+        {
+          "type": "text",
+          "id": "heading_2",
+          "default": "Most popular",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.heading_2.label"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.global.settings.header__mobile.content"
+        },
+        {
+          "type": "checkbox",
+          "id": "show_vendor",
+          "label": "t:sections.global.settings.show_product_vendor.label",
+          "default": false
+        },
+        {
+          "type": "checkbox",
+          "id": "show_price",
+          "label": "t:sections.global.settings.show_product_price.label",
+          "default": true
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu_tab.settings.header__tab_1.content"
+        },
+        {
+          "type": "text",
+          "id": "tab_target_1",
+          "default": "Tab heading",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target.label"
+        },
+        {
+          "type": "url",
+          "id": "tab_target_link_1",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target_link.label"
+        },
+        {
+          "type": "product_list",
+          "id": "tab_products_1",
+          "label": "t:sections.global.settings.products.label"
+        },
+        {
+          "type": "image_picker",
+          "id": "tab_icon_1",
+          "label": "t:sections.global.settings.icon_image.label"
+        },
+        {
+          "type": "range",
+          "id": "tab_icon_width_1",
+          "label": "t:sections.global.settings.icon_width.label",
+          "default": 42,
+          "min": 10,
+          "max": 120,
+          "step": 2,
+          "unit": "px"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu.settings.header__promo_1.content"
+        },
+        {
+          "type": "image_picker",
+          "id": "promo_image_1",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "select",
+          "id": "promo_image_ratio_1",
+          "options": [
+            {
+              "value": "adapt",
+              "label": "t:sections.global.settings.image_ratio.options__1.label"
+            },
+            {
+              "value": "square",
+              "label": "t:sections.global.settings.image_ratio.options__2.label"
+            },
+            {
+              "value": "portrait",
+              "label": "t:sections.global.settings.image_ratio.options__3.label"
+            },
+            {
+              "value": "landscape",
+              "label": "t:sections.global.settings.image_ratio.options__4.label"
+            },
+            {
+              "value": "wide",
+              "label": "t:sections.global.settings.image_ratio.options__5.label"
+            }
+          ],
+          "default": "square",
+          "label": "t:sections.global.settings.image_ratio.label",
+          "info": "t:sections.header.blocks.mega_menu_tab.settings.promo_image_ratio.info"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "text",
+          "id": "promo_heading_1",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "inline_richtext",
+          "id": "promo_subtext_1",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "checkbox",
+          "id": "promo_overlay_content_1",
+          "default": true,
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "url",
+          "id": "promo_link_1",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "text",
+          "id": "promo_tag_1",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "color",
+          "id": "promo_color_text_1",
+          "label": "t:sections.global.settings.color_text.label",
+          "default": "#ffffff"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "color",
+          "id": "promo_color_overlay_1",
+          "label": "t:sections.global.settings.color_overlay.label",
+          "default": "#000000"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_1 != blank }}",
+          "type": "range",
+          "id": "promo_overlay_opacity_1",
+          "min": 0,
+          "max": 100,
+          "step": 1,
+          "unit": "%",
+          "label": "t:sections.global.settings.overlay_opacity.label",
+          "default": 40
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu_tab.settings.header__tab_2.content"
+        },
+        {
+          "type": "text",
+          "id": "tab_target_2",
+          "default": "Tab heading",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target.label"
+        },
+        {
+          "type": "url",
+          "id": "tab_target_link_2",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target_link.label"
+        },
+        {
+          "type": "product_list",
+          "id": "tab_products_2",
+          "label": "t:sections.global.settings.products.label"
+        },
+        {
+          "type": "image_picker",
+          "id": "tab_icon_2",
+          "label": "t:sections.global.settings.icon_image.label"
+        },
+        {
+          "type": "range",
+          "id": "tab_icon_width_2",
+          "label": "t:sections.global.settings.icon_width.label",
+          "default": 42,
+          "min": 10,
+          "max": 120,
+          "step": 2,
+          "unit": "px"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu.settings.header__promo_2.content"
+        },
+        {
+          "type": "image_picker",
+          "id": "promo_image_2",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "select",
+          "id": "promo_image_ratio_2",
+          "options": [
+            {
+              "value": "adapt",
+              "label": "t:sections.global.settings.image_ratio.options__1.label"
+            },
+            {
+              "value": "square",
+              "label": "t:sections.global.settings.image_ratio.options__2.label"
+            },
+            {
+              "value": "portrait",
+              "label": "t:sections.global.settings.image_ratio.options__3.label"
+            },
+            {
+              "value": "landscape",
+              "label": "t:sections.global.settings.image_ratio.options__4.label"
+            },
+            {
+              "value": "wide",
+              "label": "t:sections.global.settings.image_ratio.options__5.label"
+            }
+          ],
+          "default": "square",
+          "label": "t:sections.global.settings.image_ratio.label",
+          "info": "t:sections.header.blocks.mega_menu_tab.settings.promo_image_ratio.info"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "text",
+          "id": "promo_heading_2",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "inline_richtext",
+          "id": "promo_subtext_2",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "checkbox",
+          "id": "promo_overlay_content_2",
+          "default": true,
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "url",
+          "id": "promo_link_2",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "text",
+          "id": "promo_tag_2",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "color",
+          "id": "promo_color_text_2",
+          "label": "t:sections.global.settings.color_text.label",
+          "default": "#ffffff"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "color",
+          "id": "promo_color_overlay_2",
+          "label": "t:sections.global.settings.color_overlay.label",
+          "default": "#000000"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_2 != blank }}",
+          "type": "range",
+          "id": "promo_overlay_opacity_2",
+          "min": 0,
+          "max": 100,
+          "step": 1,
+          "unit": "%",
+          "label": "t:sections.global.settings.overlay_opacity.label",
+          "default": 40
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu_tab.settings.header__tab_3.content"
+        },
+        {
+          "type": "text",
+          "id": "tab_target_3",
+          "default": "Tab heading",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target.label"
+        },
+        {
+          "type": "url",
+          "id": "tab_target_link_3",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target_link.label"
+        },
+        {
+          "type": "product_list",
+          "id": "tab_products_3",
+          "label": "t:sections.global.settings.products.label"
+        },
+        {
+          "type": "image_picker",
+          "id": "tab_icon_3",
+          "label": "t:sections.global.settings.icon_image.label"
+        },
+        {
+          "type": "range",
+          "id": "tab_icon_width_3",
+          "label": "t:sections.global.settings.icon_width.label",
+          "default": 42,
+          "min": 10,
+          "max": 120,
+          "step": 2,
+          "unit": "px"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu.settings.header__promo_3.content"
+        },
+        {
+          "type": "image_picker",
+          "id": "promo_image_3",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "select",
+          "id": "promo_image_ratio_3",
+          "options": [
+            {
+              "value": "adapt",
+              "label": "t:sections.global.settings.image_ratio.options__1.label"
+            },
+            {
+              "value": "square",
+              "label": "t:sections.global.settings.image_ratio.options__2.label"
+            },
+            {
+              "value": "portrait",
+              "label": "t:sections.global.settings.image_ratio.options__3.label"
+            },
+            {
+              "value": "landscape",
+              "label": "t:sections.global.settings.image_ratio.options__4.label"
+            },
+            {
+              "value": "wide",
+              "label": "t:sections.global.settings.image_ratio.options__5.label"
+            }
+          ],
+          "default": "square",
+          "label": "t:sections.global.settings.image_ratio.label",
+          "info": "t:sections.header.blocks.mega_menu_tab.settings.promo_image_ratio.info"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "text",
+          "id": "promo_heading_3",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "inline_richtext",
+          "id": "promo_subtext_3",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "checkbox",
+          "id": "promo_overlay_content_3",
+          "default": true,
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "url",
+          "id": "promo_link_3",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "text",
+          "id": "promo_tag_3",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "color",
+          "id": "promo_color_text_3",
+          "label": "t:sections.global.settings.color_text.label",
+          "default": "#ffffff"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "color",
+          "id": "promo_color_overlay_3",
+          "label": "t:sections.global.settings.color_overlay.label",
+          "default": "#000000"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_3 != blank }}",
+          "type": "range",
+          "id": "promo_overlay_opacity_3",
+          "min": 0,
+          "max": 100,
+          "step": 1,
+          "unit": "%",
+          "label": "t:sections.global.settings.overlay_opacity.label",
+          "default": 40
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu_tab.settings.header__tab_4.content"
+        },
+        {
+          "type": "text",
+          "id": "tab_target_4",
+          "default": "Tab heading",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target.label"
+        },
+        {
+          "type": "url",
+          "id": "tab_target_link_4",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target_link.label"
+        },
+        {
+          "type": "product_list",
+          "id": "tab_products_4",
+          "label": "t:sections.global.settings.products.label"
+        },
+        {
+          "type": "image_picker",
+          "id": "tab_icon_4",
+          "label": "t:sections.global.settings.icon_image.label"
+        },
+        {
+          "type": "range",
+          "id": "tab_icon_width_4",
+          "label": "t:sections.global.settings.icon_width.label",
+          "default": 42,
+          "min": 10,
+          "max": 120,
+          "step": 2,
+          "unit": "px"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu.settings.header__promo_4.content"
+        },
+        {
+          "type": "image_picker",
+          "id": "promo_image_4",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "select",
+          "id": "promo_image_ratio_4",
+          "options": [
+            {
+              "value": "adapt",
+              "label": "t:sections.global.settings.image_ratio.options__1.label"
+            },
+            {
+              "value": "square",
+              "label": "t:sections.global.settings.image_ratio.options__2.label"
+            },
+            {
+              "value": "portrait",
+              "label": "t:sections.global.settings.image_ratio.options__3.label"
+            },
+            {
+              "value": "landscape",
+              "label": "t:sections.global.settings.image_ratio.options__4.label"
+            },
+            {
+              "value": "wide",
+              "label": "t:sections.global.settings.image_ratio.options__5.label"
+            }
+          ],
+          "default": "square",
+          "label": "t:sections.global.settings.image_ratio.label",
+          "info": "t:sections.header.blocks.mega_menu_tab.settings.promo_image_ratio.info"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "text",
+          "id": "promo_heading_4",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "inline_richtext",
+          "id": "promo_subtext_4",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "checkbox",
+          "id": "promo_overlay_content_4",
+          "default": true,
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "url",
+          "id": "promo_link_4",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "text",
+          "id": "promo_tag_4",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "color",
+          "id": "promo_color_text_4",
+          "label": "t:sections.global.settings.color_text.label",
+          "default": "#ffffff"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "color",
+          "id": "promo_color_overlay_4",
+          "label": "t:sections.global.settings.color_overlay.label",
+          "default": "#000000"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_4 != blank }}",
+          "type": "range",
+          "id": "promo_overlay_opacity_4",
+          "min": 0,
+          "max": 100,
+          "step": 1,
+          "unit": "%",
+          "label": "t:sections.global.settings.overlay_opacity.label",
+          "default": 40
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu_tab.settings.header__tab_5.content"
+        },
+        {
+          "type": "text",
+          "id": "tab_target_5",
+          "default": "Tab heading",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target.label"
+        },
+        {
+          "type": "url",
+          "id": "tab_target_link_5",
+          "label": "t:sections.header.blocks.mega_menu_tab.settings.target_link.label"
+        },
+        {
+          "type": "product_list",
+          "id": "tab_products_5",
+          "label": "t:sections.global.settings.products.label"
+        },
+        {
+          "type": "image_picker",
+          "id": "tab_icon_5",
+          "label": "t:sections.global.settings.icon_image.label"
+        },
+        {
+          "type": "range",
+          "id": "tab_icon_width_5",
+          "label": "t:sections.global.settings.icon_width.label",
+          "default": 42,
+          "min": 10,
+          "max": 120,
+          "step": 2,
+          "unit": "px"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.header.blocks.mega_menu.settings.header__promo_5.content"
+        },
+        {
+          "type": "image_picker",
+          "id": "promo_image_5",
+          "label": "t:sections.header.blocks.mega_menu.settings.promo_image.label"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
+          "type": "select",
+          "id": "promo_image_ratio_5",
+          "options": [
+            {
+              "value": "adapt",
+              "label": "t:sections.global.settings.image_ratio.options__1.label"
+            },
+            {
+              "value": "square",
+              "label": "t:sections.global.settings.image_ratio.options__2.label"
+            },
+            {
+              "value": "portrait",
+              "label": "t:sections.global.settings.image_ratio.options__3.label"
+            },
+            {
+              "value": "landscape",
+              "label": "t:sections.global.settings.image_ratio.options__4.label"
+            },
+            {
+              "value": "wide",
+              "label": "t:sections.global.settings.image_ratio.options__5.label"
+            }
+          ],
+          "default": "square",
+          "label": "t:sections.global.settings.image_ratio.label",
+          "info": "t:sections.header.blocks.mega_menu_tab.settings.promo_image_ratio.info"
+        },
+        {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "text",
           "id": "promo_heading_5",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_heading.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "inline_richtext",
           "id": "promo_subtext_5",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_subtext.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "checkbox",
           "id": "promo_overlay_content_5",
           "default": true,
           "label": "t:sections.header.blocks.mega_menu.settings.promo_overlay_content.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "url",
           "id": "promo_link_5",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_link.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "text",
           "id": "promo_tag_5",
           "label": "t:sections.header.blocks.mega_menu.settings.promo_tag.label"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "color",
           "id": "promo_color_text_5",
           "label": "t:sections.global.settings.color_text.label",
           "default": "#ffffff"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "color",
           "id": "promo_color_overlay_5",
           "label": "t:sections.global.settings.color_overlay.label",
           "default": "#000000"
         },
         {
+          "visible_if": "{{ block.settings.promo_image_5 != blank }}",
           "type": "range",
           "id": "promo_overlay_opacity_5",
           "min": 0,
