diff --git a/sections/multicolumn-with-icons.liquid b/sections/multicolumn-with-icons.liquid
index 3d43c15..9a06d5b 100644
--- a/sections/multicolumn-with-icons.liquid
+++ b/sections/multicolumn-with-icons.liquid
@@ -4,14 +4,6 @@
     --section-padding-bottom: {{ section.settings.padding_bottom }}px;
     {%- render 'section-variables', section: section -%}
   }
-
-  {% if template.suffix contains 'professionals' %}
-
-    .footer-group #shopify-section-{{ section.id }} {
-      display: none;
-    }
-
-{% endif %}
 </style>

 <div class="section section--padding{% if section.settings.divider %} section--divider{% endif %}{% if section.settings.rounded %} section--rounded relative{% endif %}">
@@ -211,12 +203,14 @@
       "label": "t:sections.global.settings.button_link.label"
     },
     {
+      "visible_if": "{{ section.settings.button_label != blank }}",
       "type": "checkbox",
       "id": "button_external",
       "label": "t:sections.global.settings.button_external.label",
       "default": false
     },
     {
+      "visible_if": "{{ section.settings.button_label != blank }}",
       "type": "checkbox",
       "id": "button_icon",
       "label": "t:sections.global.settings.button_icon.label",
@@ -264,6 +258,7 @@
       "label": "t:sections.global.settings.highlighted_text.label"
     },
     {
+      "visible_if": "{{ section.settings.highlighted_text == 'scribble' }}",
       "type": "select",
       "id": "highlighted_scribble",
       "options": [
@@ -681,6 +676,7 @@
           "label": "t:sections.global.settings.highlighted_text.label"
         },
         {
+          "visible_if": "{{ block.settings.highlighted_text == 'scribble' }}",
           "type": "select",
           "id": "highlighted_scribble",
           "options": [
@@ -714,7 +710,8 @@
   ],
   "presets": [
     {
-      "name": "t:sections.multicolumn-with-icons.presets.name",
+      "name": "t:sections.multicolumn-with-icons.name",
+      "category": "t:categories.storytelling",
       "blocks": [
         {
           "type": "column",
