diff --git a/sections/timeline.liquid b/sections/timeline.liquid
index 37c90b4..f0c2390 100644
--- a/sections/timeline.liquid
+++ b/sections/timeline.liquid
@@ -30,7 +30,7 @@
     {%- render 'section-heading', section: section, section_button: indicators -%}

     {%- if section.blocks.size > 0 -%}
-      <slider-element id="Slider-{{ section.id }}" class="grid slider slider--desktop slider--tablet" selector=".card-grid>.card">
+      <slider-element id="Slider-{{ section.id }}" class="grid slider slider--desktop slider--tablet" selector=".card-grid>.card" tabindex="0">
         <motion-list class="timeline card-grid grid{% if section.blocks.size == 1 %} with-only1{% endif %}">
           {%- liquid
             assign with_gradient = false
@@ -40,25 +40,56 @@
           -%}
           {%- for block in section.blocks -%}
             <div class="timeline__item card{% if forloop.first %} selected{% endif %} relative" {{ block.shopify_attributes }}>
-              <div class="flex flex-col md:grid grid-cols-2 overflow-hidden">
-                <picture class="media media--{{ section.settings.image_ratio }} mobile:media--{{ section.settings.image_ratio_mobile }} block relative overflow-hidden">
-                  {%- if block.settings.image != blank -%}
-                    {%- assign sizes = '(max-width: 767px) 100vw, 50vw' -%}
-                    {%- if block.settings.image_mobile -%}
-                      <source
-                        media="(max-width: 767px)"
-                        srcset="{{ block.settings.image_mobile | image_url: width: '400x' }} 400w, {{ block.settings.image_mobile | image_url: width: '600x' }} 600w, {{ block.settings.image_mobile | image_url: width: '800x' }} 800w, {{ block.settings.image_mobile | image_url: width: '1000x' }} 1000w"
-                        width="{{ block.settings.image_mobile.width }}"
-                        height="{{ block.settings.image_mobile.height }}"
-                      >
+              <div class="flex flex-col md:grid{% unless section.settings.layout == 'image_first' %} md:grid-row-reverse{% endunless %} grid-cols-2 overflow-hidden">
+                {%- if block.type == 'content' -%}
+                  <picture class="media media--{{ section.settings.image_ratio }} mobile:media--{{ section.settings.image_ratio_mobile }} block relative overflow-hidden">
+                    {%- if block.settings.image != blank -%}
+                      {%- assign sizes = '(max-width: 767px) 100vw, 50vw' -%}
+                      {%- if block.settings.image_mobile -%}
+                        <source
+                          media="(max-width: 767px)"
+                          srcset="{{ block.settings.image_mobile | image_url: width: '400x' }} 400w, {{ block.settings.image_mobile | image_url: width: '600x' }} 600w, {{ block.settings.image_mobile | image_url: width: '800x' }} 800w, {{ block.settings.image_mobile | image_url: width: '1000x' }} 1000w"
+                          width="{{ block.settings.image_mobile.width }}"
+                          height="{{ block.settings.image_mobile.height }}"
+                        >
+                      {%- endif -%}
+
+                      {{- block.settings.image | image_url: width: block.settings.image.width | image_tag: loading: 'lazy', sizes: sizes, widths: '300,400,500,600,700,800,900,1000,1200,1400,1600', is: 'lazy-image' -}}
+                    {%- else -%}
+                      {%- capture placeholder -%}{% cycle 'collection-1', 'collection-2', 'collection-3', 'collection-4', 'collection-5', 'collection-6' %}{%- endcapture -%}
+                      {{- placeholder | placeholder_svg_tag: 'placeholder placeholder--transparent' -}}
+                    {%- endif -%}
+                  </picture>
+                {%- else -%}
+                  <div class="media media--{{ section.settings.image_ratio }} mobile:media--{{ section.settings.image_ratio_mobile }} block relative overflow-hidden">
+                    {%- if block.settings.video_mobile or block.settings.video_url_mobile -%}
+                      {%- render 'video',
+                        video: block.settings.video_mobile,
+                        video_url: block.settings.video_url_mobile,
+                        muted: false,
+                        autoplay: false,
+                        loop: false,
+                        description: block.settings.description,
+                        cover_image: block.settings.cover_image_mobile,
+                        show_play_button: true,
+                        is_mobile: true,
+                        class: 'media w-full h-full'
+                      -%}
                     {%- endif -%}

-                    {{- block.settings.image | image_url: width: block.settings.image.width | image_tag: loading: 'lazy', sizes: sizes, widths: '300,400,500,600,700,800,900,1000,1200,1400,1600', is: 'lazy-image' -}}
-                  {%- else -%}
-                    {%- capture placeholder -%}{% cycle 'collection-1', 'collection-2', 'collection-3', 'collection-4', 'collection-5', 'collection-6' %}{%- endcapture -%}
-                    {{- placeholder | placeholder_svg_tag: 'placeholder' -}}
-                  {%- endif -%}
-                </picture>
+                    {%- render 'video',
+                      video: block.settings.video,
+                      video_url: block.settings.video_url,
+                      muted: false,
+                      autoplay: false,
+                      loop: false,
+                      description: block.settings.description,
+                      cover_image: block.settings.cover_image,
+                      show_play_button: true,
+                      class: 'media w-full h-full'
+                    -%}
+                  </div>
+                {%- endif -%}
                 <div class="timeline__item-content flex flex-col gap-4">
                   {%- if block.settings.subheading != blank -%}
                     <p class="text-opacity text-base lg:text-lg font-bold tracking-tight">{{ block.settings.subheading | escape }}</p>
@@ -67,7 +98,7 @@
                   {%- if block.settings.heading != blank -%}
                     <{{ block.settings.heading_tag }} class="heading lg:text-3xl text-2xl leading-none tracking-tight">
                       {%- if block.settings.highlighted_text == 'none' -%}
-                        <split-words class="split-words block" data-animate="fade-up-large">{{ block.settings.heading }}</split-words>
+                        <split-words class="split-words js-invisible block" data-animate="fade-up-large">{{ block.settings.heading }}</split-words>
                       {%- else -%}
                         {%- render 'highlighted-text', content: block.settings.heading, style: block.settings.highlighted_text, scribble: block.settings.highlighted_scribble, gradient: with_gradient -%}
                       {%- endif -%}
@@ -100,20 +131,30 @@

       {%- if section.blocks.size > 1 -%}
         <div class="scroll-area">
-          <slider-dots class="timeline-dots grid gap-2d5" align-selected=".scroll-area" aria-controls="Slider-{{ section.id }}">
-            {%- for block in section.blocks -%}
-              <button type="button" class="heading text-lg text-left flex items-center gap-2d5" data-index="{{ forloop.index }}" aria-current="{% if forloop.first %}true{% else %}false{% endif %}">
-                <span class="sr-only">{{ 'general.pagination.page' | t: number: forloop.index }}</span>
-                {%- liquid
-                  if block.settings.subheading != blank
-                    echo block.settings.subheading | escape
-                  else
-                    echo forloop.index
-                  endif
-                -%}
-              </button>
-            {%- endfor -%}
-          </slider-dots>
+          {%- if section.settings.show_navigation -%}
+            <slider-dots class="timeline-dots grid gap-2d5" align-selected=".scroll-area" aria-controls="Slider-{{ section.id }}">
+              {%- for block in section.blocks -%}
+                <button type="button" class="heading text-lg text-left flex items-center gap-2d5" data-index="{{ forloop.index }}" aria-current="{% if forloop.first %}true{% else %}false{% endif %}">
+                  <span class="sr-only">{{ 'general.pagination.page' | t: number: forloop.index }}</span>
+                  {%- liquid
+                    if block.settings.subheading != blank
+                      echo block.settings.subheading | escape
+                    else
+                      echo forloop.index
+                    endif
+                  -%}
+                </button>
+              {%- endfor -%}
+            </slider-dots>
+          {%- else -%}
+            <slider-dots class="flickity-page-dots lg:hidden" align-selected=".scroll-area" aria-controls="Slider-{{ section.id }}">
+              {%- for block in section.blocks -%}
+                <button type="button" class="flickity-page-dot" data-index="{{ forloop.index }}" aria-current="{% if forloop.first %}true{% else %}false{% endif %}">
+                  <span class="sr-only">{{ 'general.pagination.page' | t: number: forloop.index }}</span>
+                </button>
+              {%- endfor -%}
+            </slider-dots>
+          {%- endif -%}
         </div>
       {%- endif -%}
     {%- endif -%}
@@ -125,6 +166,29 @@
   "name": "t:sections.timeline.name",
   "max_blocks": 15,
   "settings": [
+    {
+      "type": "checkbox",
+      "id": "show_navigation",
+      "label": "t:sections.timeline.settings.show_navigation.label",
+      "default": true
+    },
+    {
+      "type": "select",
+      "id": "layout",
+      "options": [
+        {
+          "value": "image_first",
+          "label": "t:sections.timeline.settings.layout.options__1.label"
+        },
+        {
+          "value": "text_first",
+          "label": "t:sections.timeline.settings.layout.options__2.label"
+        }
+      ],
+      "default": "image_first",
+      "label": "t:sections.timeline.settings.layout.label",
+      "info": "t:sections.timeline.settings.layout.info"
+    },
     {
       "type": "select",
       "id": "image_ratio",
@@ -332,6 +396,7 @@
       "label": "t:sections.global.settings.highlighted_text.label"
     },
     {
+      "visible_if": "{{ section.settings.highlighted_text == 'scribble' }}",
       "type": "select",
       "id": "highlighted_scribble",
       "options": [
@@ -463,6 +528,256 @@
           "id": "image_mobile",
           "label": "t:sections.global.settings.image_mobile.label"
         },
+        {
+          "type": "header",
+          "content": "t:sections.global.settings.header__content.content"
+        },
+        {
+          "type": "text",
+          "id": "subheading",
+          "label": "t:sections.global.settings.subheading.label",
+          "default": "Text"
+        },
+        {
+          "type": "inline_richtext",
+          "id": "heading",
+          "label": "t:sections.global.settings.heading.label",
+          "default": "Heading"
+        },
+        {
+          "type": "select",
+          "id": "heading_tag",
+          "options": [
+            {
+              "value": "h1",
+              "label": "t:sections.global.settings.heading_tag.options__1.label"
+            },
+            {
+              "value": "h2",
+              "label": "t:sections.global.settings.heading_tag.options__2.label"
+            },
+            {
+              "value": "h3",
+              "label": "t:sections.global.settings.heading_tag.options__3.label"
+            },
+            {
+              "value": "h4",
+              "label": "t:sections.global.settings.heading_tag.options__4.label"
+            },
+            {
+              "value": "h5",
+              "label": "t:sections.global.settings.heading_tag.options__5.label"
+            },
+            {
+              "value": "h6",
+              "label": "t:sections.global.settings.heading_tag.options__6.label"
+            },
+            {
+              "value": "div",
+              "label": "t:sections.global.settings.heading_tag.options__7.label"
+            },
+            {
+              "value": "span",
+              "label": "t:sections.global.settings.heading_tag.options__8.label"
+            },
+            {
+              "value": "p",
+              "label": "t:sections.global.settings.heading_tag.options__9.label"
+            }
+          ],
+          "default": "h2",
+          "label": "t:sections.global.settings.heading_tag.label",
+          "info": "t:sections.global.settings.heading_tag.info"
+        },
+        {
+          "type": "richtext",
+          "id": "text",
+          "label": "t:sections.global.settings.text.label",
+          "default": "<p>Use this text to share information about your brand with your customers. Describe a product, share announcements, or welcome customers to your store.</p>"
+        },
+        {
+          "type": "text",
+          "id": "button_label",
+          "label": "t:sections.global.settings.button_label.label",
+          "info": "t:sections.global.settings.button_label.info"
+        },
+        {
+          "type": "url",
+          "id": "button_link",
+          "label": "t:sections.global.settings.button_link.label"
+        },
+        {
+          "visible_if": "{{ block.settings.button_label != blank }}",
+          "type": "select",
+          "id": "button_style",
+          "options": [
+            {
+              "value": "primary",
+              "label": "t:sections.global.settings.button_style.options__1.label"
+            },
+            {
+              "value": "secondary",
+              "label": "t:sections.global.settings.button_style.options__2.label"
+            },
+            {
+              "value": "link",
+              "label": "t:sections.global.settings.button_style.options__3.label"
+            }
+          ],
+          "default": "primary",
+          "label": "t:sections.global.settings.button_style.label"
+        },
+        {
+          "visible_if": "{{ block.settings.button_label != blank }}",
+          "type": "checkbox",
+          "id": "button_external",
+          "label": "t:sections.global.settings.button_external.label",
+          "default": false
+        },
+        {
+          "visible_if": "{{ block.settings.button_label != blank }}",
+          "type": "checkbox",
+          "id": "button_icon",
+          "label": "t:sections.global.settings.button_icon.label",
+          "default": true
+        },
+        {
+          "type": "header",
+          "content": "t:sections.global.settings.header__highlight.content",
+          "info": "t:sections.global.settings.header__highlight.info"
+        },
+        {
+          "type": "select",
+          "id": "highlighted_text",
+          "options": [
+            {
+              "value": "none",
+              "label": "t:sections.global.settings.highlighted_text.options__1.label"
+            },
+            {
+              "value": "text",
+              "label": "t:sections.global.settings.highlighted_text.options__2.label"
+            },
+            {
+              "value": "full_text",
+              "label": "t:sections.global.settings.highlighted_text.options__3.label"
+            },
+            {
+              "value": "half_text",
+              "label": "t:sections.global.settings.highlighted_text.options__4.label"
+            },
+            {
+              "value": "underline",
+              "label": "t:sections.global.settings.highlighted_text.options__5.label"
+            },
+            {
+              "value": "stencil",
+              "label": "t:sections.global.settings.highlighted_text.options__6.label"
+            },
+            {
+              "value": "scribble",
+              "label": "t:sections.global.settings.highlighted_text.options__7.label"
+            }
+          ],
+          "default": "none",
+          "label": "t:sections.global.settings.highlighted_text.label"
+        },
+        {
+          "visible_if": "{{ block.settings.highlighted_text == 'scribble' }}",
+          "type": "select",
+          "id": "highlighted_scribble",
+          "options": [
+            {
+              "value": "circle",
+              "label": "t:sections.global.settings.highlighted_scribble.options__1.label"
+            },
+            {
+              "value": "basic-underline",
+              "label": "t:sections.global.settings.highlighted_scribble.options__2.label"
+            },
+            {
+              "value": "sketch-underline",
+              "label": "t:sections.global.settings.highlighted_scribble.options__3.label"
+            },
+            {
+              "value": "squiggle-underline",
+              "label": "t:sections.global.settings.highlighted_scribble.options__4.label"
+            },
+            {
+              "value": "squiggle-underline-2",
+              "label": "t:sections.global.settings.highlighted_scribble.options__5.label"
+            }
+          ],
+          "default": "circle",
+          "label": "t:sections.global.settings.highlighted_scribble.label",
+          "info": "t:sections.global.settings.highlighted_scribble.info"
+        }
+      ]
+    },
+    {
+      "type": "content_video",
+      "name": "t:sections.timeline.blocks.content_video.name",
+      "settings": [
+        {
+          "type": "header",
+          "content": "t:sections.video.settings.header__video.content"
+        },
+        {
+          "type": "video",
+          "id": "video",
+          "label": "t:sections.video.settings.video.label"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.video.settings.header__video_url.content",
+          "info": "t:sections.video.settings.header__video_url.info"
+        },
+        {
+          "type": "video_url",
+          "id": "video_url",
+          "accept": ["youtube", "vimeo"],
+          "default": "https://www.youtube.com/watch?v=_9VUPq3SxOc",
+          "label": "t:sections.video.settings.video_url.label",
+          "info": "t:sections.video.settings.video_url.info"
+        },
+        {
+          "visible_if": "{{ block.settings.video_url != blank }}",
+          "type": "image_picker",
+          "id": "cover_image",
+          "label": "t:sections.video.settings.cover_image.label"
+        },
+        {
+          "type": "text",
+          "id": "description",
+          "label": "t:sections.video.settings.description.label",
+          "info": "t:sections.video.settings.description.info"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.global.settings.header__mobile.content"
+        },
+        {
+          "type": "video",
+          "id": "video_mobile",
+          "label": "t:sections.video.settings.video_mobile.label"
+        },
+        {
+          "type": "video_url",
+          "id": "video_url_mobile",
+          "accept": ["youtube", "vimeo"],
+          "label": "t:sections.video.settings.video_url_mobile.label",
+          "info": "t:sections.video.settings.video_url.info"
+        },
+        {
+          "visible_if": "{{ block.settings.video_url_mobile != blank }}",
+          "type": "image_picker",
+          "id": "cover_image_mobile",
+          "label": "t:sections.video.settings.cover_image_mobile.label"
+        },
+        {
+          "type": "header",
+          "content": "t:sections.global.settings.header__content.content"
+        },
         {
           "type": "text",
           "id": "subheading",
@@ -538,6 +853,7 @@
           "label": "t:sections.global.settings.button_link.label"
         },
         {
+          "visible_if": "{{ block.settings.button_label != blank }}",
           "type": "select",
           "id": "button_style",
           "options": [
@@ -558,12 +874,14 @@
           "label": "t:sections.global.settings.button_style.label"
         },
         {
+          "visible_if": "{{ block.settings.button_label != blank }}",
           "type": "checkbox",
           "id": "button_external",
           "label": "t:sections.global.settings.button_external.label",
           "default": false
         },
         {
+          "visible_if": "{{ block.settings.button_label != blank }}",
           "type": "checkbox",
           "id": "button_icon",
           "label": "t:sections.global.settings.button_icon.label",
@@ -611,6 +929,7 @@
           "label": "t:sections.global.settings.highlighted_text.label"
         },
         {
+          "visible_if": "{{ block.settings.highlighted_text == 'scribble' }}",
           "type": "select",
           "id": "highlighted_scribble",
           "options": [
@@ -644,22 +963,43 @@
   ],
   "presets": [
     {
-      "name": "t:sections.timeline.presets.name",
+      "name": "t:sections.timeline.name",
+      "category": "t:categories.brand_trust",
       "blocks": [
         {
-          "type": "content"
+          "type": "content",
+          "settings": {
+            "subheading": "2020",
+            "heading": "Example heading"
+          }
         },
         {
-          "type": "content"
+          "type": "content",
+          "settings": {
+            "subheading": "2021",
+            "heading": "Example heading"
+          }
         },
         {
-          "type": "content"
+          "type": "content",
+          "settings": {
+            "subheading": "2023",
+            "heading": "Example heading"
+          }
         },
         {
-          "type": "content"
+          "type": "content",
+          "settings": {
+            "subheading": "2024",
+            "heading": "Example heading"
+          }
         },
         {
-          "type": "content"
+          "type": "content",
+          "settings": {
+            "subheading": "2025",
+            "heading": "Example heading"
+          }
         }
       ]
     }
